apps/microsoft-teams/frontend/src/components/config/SearchableList/SearchableList.tsx

```diff
@@ -0,0 +1,56 @@
+import Fuse from 'fuse.js';
+import { ReactNode, useCallback, useEffect, useState } from 'react';
+import React from 'react';
+
+interface Props<T> {
+  list: T[];
+  searchQuery: string;
+  renderListItem: (item: T) => ReactNode;
+  searchKeys: string[];
+}
+
+/**
+ * @description - Higher order component to make a list fuzzy-searchable with fuse.js.  The actual
+ * implementation of the search options, as well as the rendering of filtered items is left up to
+ * the consumer of this component.
+ */
+const SearchableList = <T,>({ list, searchQuery, searchKeys, renderListItem }: Props<T>) => {
+  const [filteredList, setFilteredList] = useState<T[]>(list);
+
+  const filterList = useCallback((searchPattern: string) => {
+    if (searchPattern === '') {
+      // revert to default full list of available contentTypes
+      setFilteredList(list);
+      return;
+    }
+
+    const fuseOptions = {
+      isCaseSensitive: false,
+      keys: searchKeys,
+    };
+
+    const fuse = new Fuse(list, fuseOptions);
+    // const searchPattern = e.target.value;
```

Comment:
You probably meant to delete this commented out code?

Reply:
Yep, this was a carry-over from when this was more tightly coupled to a search input, as opposed to just passing `searchQuery`.  Good call.

- [ ] remove comment